cdc/entry/schema_storage_test.go

```diff
@@ -770,3 +777,195 @@ func (t *schemaSuite) TestExplicitTables(c *check.C) {
+			lname VARCHAR(25) NOT NULL,
+			store_id INT NOT NULL,
+			department_id INT NOT NULL
+		)
+
+		PARTITION BY RANGE(id)  (
+			PARTITION p0 VALUES LESS THAN (5),
+			PARTITION p1 VALUES LESS THAN (10),
+			PARTITION p2 VALUES LESS THAN (15),
+			PARTITION p3 VALUES LESS THAN (20)
+		)`, // ActionCreateTable
+		"ALTER TABLE test_ddl2.employees DROP PARTITION p2",                                  // ActionDropTablePartition
+		"ALTER TABLE test_ddl2.employees ADD PARTITION (PARTITION p4 VALUES LESS THAN (25))", //ActionAddTablePartition
+		"ALTER TABLE test_ddl2.employees TRUNCATE PARTITION p3",                              // ActionTruncateTablePartition
+		"alter table test_ddl2.employees comment='modify comment'",                           // ActionModifyTableComment
+		"alter table test_ddl2.simple_test1 drop primary key",                                // ActionDropPrimaryKey
+		"alter table test_ddl2.simple_test1 add primary key pk(id)",                          // ActionAddPrimaryKey
+		"ALTER TABLE test_ddl2.simple_test1 ALTER id SET DEFAULT 18",                         // ActionSetDefaultValue
+		"ALTER TABLE test_ddl2.simple_test1 CHARACTER SET = utf8mb4",                         // ActionModifyTableCharsetAndCollate
+		// "recover table test_ddl2.employees",                                                  // ActionRecoverTable this ddl can't work on mock tikv
+	}}
+
+	checkSnapsEquals := func(snapA *schemaSnapshot, snapB *schemaSnapshot) {
+		c.Assert(snapA, check.DeepEquals, snapB,
+			check.Commentf("%s", cmp.Diff(snapA, snapB, cmp.AllowUnexported(schemaSnapshot{}, model.TableInfo{}))))
+	}
+
+	testOneGroup := func(tc []string) {
+		store, err := mockstore.NewMockStore()
+		c.Assert(err, check.IsNil)
+		defer store.Close() //nolint:errcheck
+		ticonfig.UpdateGlobal(func(conf *ticonfig.Config) {
+			conf.AlterPrimaryKey = true
+		})
+		session.SetSchemaLease(0)
+		session.DisableStats4Test()
+		domain, err := session.BootstrapSession(store)
+		c.Assert(err, check.IsNil)
+		defer domain.Close()
+		domain.SetStatsUpdating(true)
+		tk := testkit.NewTestKit(c, store)
+
+		for _, ddlSQL := range tc {
+			tk.MustExec(ddlSQL)
+		}
+
+		jobs, err := getAllHistoryDDLJob(store)
+		c.Assert(err, check.IsNil)
+		scheamStorage, err := NewSchemaStorage(nil, 0, nil, false)
+		c.Assert(err, check.IsNil)
+		for _, job := range jobs {
+			err := scheamStorage.HandleDDLJob(job)
+			c.Assert(err, check.IsNil)
+		}
+
+		for _, job := range jobs {
+			ts := job.BinlogInfo.FinishedTS
+			meta, err := kv.GetSnapshotMeta(store, ts)
+			c.Assert(err, check.IsNil)
+			snapFromMeta, err := newSchemaSnapshotFromMeta(meta, ts, false)
+			c.Assert(err, check.IsNil)
+			snapFromSchemaStore, err := scheamStorage.GetSnapshot(ctx, ts)
+			c.Assert(err, check.IsNil)
+
+			tidySchemaSnapshot(snapFromMeta)
+			tidySchemaSnapshot(snapFromSchemaStore)
+			// check if the two snapshot are equal.
+			checkSnapsEquals(snapFromMeta, snapFromSchemaStore)
+		}
```

Comment:
Could we also check the historical snapshot from the `schemaStorage`, such as querying by `ts-1` or last job FinishedTs